### PR TITLE
fix(ssr): ensure useRequestHeaders are case-insensitive

### PR DESCRIPTION
**Pull request by manniL**
### 🔗 Linked issue

https://github.com/nuxt/framework/discussions/8803

### ❓ Type of change

- [ ] 📖 Documentation (updates to the documentation or readme)
- [x] 🐞 Bug fix (a non-breaking change that fixes an issue)
- [ ] 👌 Enhancement (improving an existing functionality like performance)
- [ ] ✨ New feature (a non-breaking change that adds functionality)
- [ ] ⚠️ Breaking change (fix or feature that would cause existing functionality to change)

### 📚 Description

Makes headers case-insensitive.

Would be good to double-check if headers always come lower-cased from the server. If no, maybe add a transform step there too.
https://github.com/unjs/h3/blob/6beee463d0e95207480acfd21badc2c2fa46f6c3/src/utils/request.ts#L58-L65 does not handle casing. Should I create a PR or an issue there too?

### 📝 Checklist

- [x] I have linked an issue or discussion.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: packages/nuxt/src/app/composables/ssr.ts [hunk 1]
```diff
@@ -8,7 +8,7 @@ export function useRequestHeaders (include?: any[]) {
   if (process.client) { return {} }
   const headers = useNuxtApp().ssrContext?.event.req.headers ?? {}
   if (!include) { return headers }
-  return Object.fromEntries(include.filter(key => headers[key]).map(key => [key, headers[key]]))
+  return Object.fromEntries(include.map(key => key.toLowerCase()).filter(key => headers[key]).map(key => [key, headers[key]]))
```
Comment (aaharu) [inline packages/nuxt/src/app/composables/ssr.ts:11]: useRequestHeaders's return type is `Record<K, string | undefined>`.

In this case, object key is not `K`.
```ts
useRequestHeaders<'Cookie'>(['Cookie'])
// returns { cookie: '...' }
```

Comment (manniL) [reply packages/nuxt/src/app/composables/ssr.ts:11]: Good point! Do you have a suggestion how to avoid this?

Comment (danielroe) [reply packages/nuxt/src/app/composables/ssr.ts:11]: pushed lowercase type helper.

**Comment by manniL** [comment]
Oh well, that was easy. Thanks @danielroe ☺️

**Review by pi0 — APPROVED** [review]
LGTM. We might also update this utility with new `h3.useRequestHeaders` that does the normalization

**Pull request merged**